### PR TITLE
drivers: stm32: Convert STM32 drivers to new DT_INST macros

### PR DESCRIPTION
**Pull request by galak**
Convert older DT_INST_ macro use in STM32 drivers to the new
include/devicetree.h DT_INST macro APIs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: drivers/usb/device/usb_dc_stm32.c [hunk 1]
```diff
@@ -60,34 +62,34 @@ LOG_MODULE_REGISTER(usb_dc_stm32);
...
+#define DT_USB_BASE_ADDRESS		DT_REG_ADDR(DT_INST(0, st_stm32_usb))
+#define DT_USB_IRQ			DT_IRQ_BY_NAME(DT_INST(0, st_stm32_usb), usb, irq)
+#define DT_USB_IRQ_PRI			DT_IRQ_BY_NAME(DT_INST(0, st_stm32_usb), usb, priority)
+#define DT_USB_NUM_BIDIR_ENDPOINTS	DT_PROP(DT_INST(0, st_stm32_usb), num_bidir_endpoints)
+#define DT_USB_RAM_SIZE			DT_PROP(DT_INST(0, st_stm32_usb), ram_size)
+#define DT_USB_MAXIMUM_SPEED		DT_PROP(DT_INST(0, st_stm32_usb), maximum_speed)
+#define DT_USB_CLOCK_BITS		DT_CLOCKS_CELL(DT_INST(0, st_stm32_usb), bits)
+#define DT_USB_CLOCK_BUS		DT_CLOCKS_CELL(DT_INST(0, st_stm32_usb), bus)
+#if DT_NODE_HAS_PROP(DT_INST(0, st_stm32_usb), enable_pin_remap)
+#define DT_USB_ENABLE_PIN_REMAP		DT_PROP(DT_INST(0, st_stm32_usb), enable_pin_remap)
```
Comment (erwango) [inline drivers/usb/device/usb_dc_stm32.c]: Is that working ?
These compats are mutually exclusives, I'm not sure you'll get it work that way.

**Review by erwango — COMMENTED** [review]
Quick first feedback

File: drivers/usb/device/usb_dc_stm32.c (hunk 1 above)
Comment (galak) [reply drivers/usb/device/usb_dc_stm32.c]: Why not?  The specific compat (`st_stm32_usb`, `st_stm32_otgfs` or `st_stm32_otghs`) is explicit specified in the macro.

**Review by gmarull — APPROVED** [review]

**Review by ydamigos — APPROVED** [review]
Looks good to me

File: drivers/usb/device/usb_dc_stm32.c (hunk 1 above)
Comment (erwango) [reply drivers/usb/device/usb_dc_stm32.c]: ok, got it. Will take more time to review this more carefully.

Comment (galak) [reply drivers/usb/device/usb_dc_stm32.c]: thanks

**Review by erwango — APPROVED** [review]

**Pull request merged**